cmake/RunReverseDD.cmake

```diff
@@ -0,0 +1,201 @@
+        # Find and run basement
+        find_program(BASEMENT_EXE Basement PATHS "${PRODUCT_PATH}" NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH NO_SYSTEM_ENVIRONMENT_PATH
+                                                                   NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
+        message("Executing Basement from ${BASEMENT_EXE}")
+        execute_process(COMMAND "${BASEMENT_EXE}" WORKING_DIRECTORY "${OUTPUT_DIR_PATH}")
+        # Then copy basement results into the expanded file
+        file(READ "${OUTPUT_DIR_PATH}/EPObjects.TXT" BASEMENT_CONTENTS)
+        file(APPEND "${OUTPUT_DIR_PATH}/expanded.idf" "${BASEMENT_CONTENTS}")
+      endif()
+
+      set(IDF_PATH "${OUTPUT_DIR_PATH}/expanded.idf")
+
+    endif() # expand objects found something and created expanded.idf
+
+    list(FIND ENERGYPLUS_FLAGS_LIST -x EXPAND_RESULT)
+
+    if("${EXPAND_RESULT}" GREATER -1)
+      find_program(EXPANDOBJECTS_EXE ExpandObjects PATHS "${PRODUCT_PATH}" NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH
+                                                                           NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
+      # Move to executable directory
+      execute_process(COMMAND ${CMAKE_COMMAND} -E copy_if_different "${EXPANDOBJECTS_EXE}" "${EXE_PATH}")
+    endif()
+
+    list(FIND ENERGYPLUS_FLAGS_LIST -r READVARS_RESULT)
+
+    if("${READVARS_RESULT}" GREATER -1)
+      find_program(READVARS_EXE ReadVarsESO PATHS "${PRODUCT_PATH}" NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH NO_SYSTEM_ENVIRONMENT_PATH
+                                                                    NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
+      # Move to executable directory
+      execute_process(COMMAND ${CMAKE_COMMAND} -E copy_if_different "${READVARS_EXE}" "${EXE_PATH}")
+    endif()
+
+  endif() # build fortran
+
+  if(WIN32)
+    set(ECHO_CMD cmd /C echo.)
+  else()
+    set(ECHO_CMD "echo")
+  endif()
+
+  # set the reverse design day environment variable for EnergyPlus
+  set(ENV{REVERSEDD} "${V}")
+  execute_process(
+    COMMAND ${ECHO_CMD}
+    COMMAND "${ENERGYPLUS_EXE}" -w "${EPW_PATH}" -d "${OUTPUT_DIR_PATH}" ${ENERGYPLUS_FLAGS_LIST} "${IDF_PATH}"
+    WORKING_DIRECTORY "${OUTPUT_DIR_PATH}"
+    RESULT_VARIABLE RESULT)
+
+  if (RESULT EQUAL 0)
+    # pass
+  else()
+    # if there was a problem with E+, it will either be caught with integration tests, or it is because of a problem
+    # with reverse DD run ... such as zero design days ... so I just won't catch those for now
+    message("Test Passed")
+    return()
+  endif()
+endforeach()
+
+message("Reverse design day simulations completed, now running comparison script")
+
+execute_process(
+        COMMAND ${ECHO_CMD}
+        COMMAND "${PYTHON_EXECUTABLE}" "${SOURCE_DIR}/cmake/ReverseDDPostProcess.py" "${BINARY_DIR}/${OUTPUT_DIR_NAME}/${IDF_NAME}"
+        WORKING_DIRECTORY "${OUTPUT_DIR_PATH}"
+        RESULT_VARIABLE RESULT_RDD)
```

Comment:
Once EnergyPlus regular and reversed are completed, we call the new Python comparison script to evaluate whether the files are meaningfully different or not.  The Python script exit code will be used as the signal.